src/contractor/contractor.cpp

```diff
@@ -366,36 +367,56 @@ EdgeID Contractor::LoadEdgeExpandedGraph(
         throw util::exception("Limit of 255 segment speed and turn penalty files each reached");
 
     util::SimpleLogger().Write() << "Opening " << edge_based_graph_filename;
-    boost::filesystem::ifstream input_stream(edge_based_graph_filename, std::ios::binary);
-    if (!input_stream)
-        throw util::exception("Could not load edge based graph file");
+
+    const boost::interprocess::file_mapping edge_based_graph_mapping{
+        edge_based_graph_filename.c_str(), boost::interprocess::read_only};
```

Comment:
The following code is duplicated at least three times. Can we abstract over it providing a lambda / function that takes a file name and a madvise hint, and returns a region?